Storage/composer.json

```diff
@@ -4,7 +4,8 @@
     "license": "Apache-2.0",
     "minimum-stability": "stable",
     "require": {
-        "google/cloud-core": "^1.23"
+        "google/cloud-core": "^1.23",
```

Comment:
This is a reminder for me: This will need to be updated to require at least the latest version of `cloud-core`, which should be `1.25`. Need an elegant way to do this, which may require breaking the core changes in this PR out into their own PR.